gateway-messages/src/sp_impl.rs

```diff
@@ -515,15 +518,19 @@ where
 /// Read the request header from the front of `data`, returning the message ID
 /// we should use in our response (pulled from the header if possible, or a
 /// sentinel if not) and either the remainder of the request data or an error.
-fn read_request_header(data: &[u8]) -> (u32, Result<&[u8], SpError>) {
+fn read_request_header(data: &[u8]) -> (Header, Result<&[u8], SpError>) {
```

Comment:
This could be a _little_ more strongly typed:
```suggestion
fn read_request_header(data: &[u8]) -> (Option<Header>, Result<&[u8], SpError>) {
```
(then, the caller would get to decide what fake message ID to use)